providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -630,4 +635,13 @@ public void restartEventStream() throws Exception {
 
         this.handleEvents();
     }
+
+    /**
+     * Call .wait() on this to block until the event stream is alive.
+     *
+     * @return eventStreamAliveSync
+     */
+    public Object getEventStreamAliveSync() {
+        return this.eventStreamAliveSync;
+    }
```

Comment:
There is already a check that the event stream is alive before retrieving from cache: https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java#L192

Reply:
OK so correct my if I'm wrong: 

The race was just really in tests - because the tests could be flaky if evaluations had started but the connection hadn't yet been made - it wouldn't be an issue in production, because the end result would be the cache just takes a few moments to start up... but that breaks the test assumptions since the 2nd eval might not be cached.

Is that right?

Reply:
That's exactly right. A similar problem occurred in the Go implementation, I went back and exposed a similar block until ready

Reply:
Great, makes sense to me. Thanks!